django_rq/views.py

```diff
@@ -4,64 +4,36 @@
 
 from django.contrib import messages
 from django.contrib.admin.views.decorators import staff_member_required
-from django.http import Http404
+from django.http import Http404, JsonResponse
 from django.shortcuts import redirect, render
 
 from redis.exceptions import ResponseError
-from rq import requeue_job, Worker
+from rq import requeue_job
 from rq.exceptions import NoSuchJobError
 from rq.job import Job
 from rq.registry import (DeferredJobRegistry, FinishedJobRegistry,
                          StartedJobRegistry)
 from rq.worker import Worker
 
-from .queues import get_connection, get_queue_by_index
-from .settings import QUEUES_LIST
-from .workers import collect_workers_by_connection, get_all_workers_by_configuration
+from .queues import get_queue_by_index
+from .settings import STATS_TOKEN
+from .utils import get_statistics
 
 
 @staff_member_required
 def stats(request):
-    queues = []
-    workers_collections = collect_workers_by_connection(QUEUES_LIST)
-    for index, config in enumerate(QUEUES_LIST):
-
-        queue = get_queue_by_index(index)
-        connection = queue.connection
-
-        queue_data = {
-            'name': queue.name,
-            'jobs': queue.count,
-            'index': index,
-            'connection_kwargs': connection.connection_pool.connection_kwargs
-        }
-
-        if queue.name == 'failed':
-            queue_data['workers'] = '-'
-            queue_data['finished_jobs'] = '-'
-            queue_data['started_jobs'] = '-'
-            queue_data['deferred_jobs'] = '-'
-
-        else:
-            connection = get_connection(queue.name)
-            all_workers = get_all_workers_by_configuration(
-                config['connection_config'],
-                workers_collections
-            )
-            queue_workers = [worker for worker in all_workers if queue in worker.queues]
-            queue_data['workers'] = len(queue_workers)
-
-            finished_job_registry = FinishedJobRegistry(queue.name, connection)
-            started_job_registry = StartedJobRegistry(queue.name, connection)
-            deferred_job_registry = DeferredJobRegistry(queue.name, connection)
-            queue_data['finished_jobs'] = len(finished_job_registry)
-            queue_data['started_jobs'] = len(started_job_registry)
-            queue_data['deferred_jobs'] = len(deferred_job_registry)
-
-        queues.append(queue_data)
-
-    context_data = {'queues': queues}
-    return render(request, 'django_rq/stats.html', context_data)
+    return render(request, 'django_rq/stats.html', get_statistics())
+
+
+def token_stats_json(request, token):
+    if token != STATS_TOKEN:
+        raise Http404
```

Comment:
Return a descriptive error message: `{"description": "Please configure API_TOKEN in settings.py before accessing this view"}`